packages/core/test/features/st-var.spec.ts

```diff
@@ -0,0 +1,1219 @@
+                prop: value(quotes);
+            }
+        `);
+
+        const { meta, exports } = sheets['/entry.st.css'];
+
+        shouldReportNoDiagnostics(meta);
+
+        // symbols
+        expect(STVar.get(meta, `double`), `double symbol`).to.contain({
+            text: `"double"`,
+        });
+        expect(STVar.get(meta, `single`), `single symbol`).to.contain({
+            text: `'single'`,
+        });
+        expect(STVar.get(meta, `quotes`), `quotes symbol`).to.contain({
+            text: `"'with-quotes'"`,
+        });
+
+        // exports - ToDo: fix outer quotes removal
+        expect(exports.stVars, `JS exports`).to.eql({
+            double: `"double"`,
+            single: `'single'`,
+            quotes: `"'with-quotes'"`,
+        });
+    });
+    it(`should resolve value in :vars definition`, () => {
+        const { sheets } = testStylableCore(`
+            /* @transform-remove */
+            :vars {
+                varA: a-val;
+                varB: value(varA);
+            }
+            .root {
+                /* @decl prop: a-val */
+                prop: value(varB);
+            }
+        `);
+
+        const { meta, exports } = sheets['/entry.st.css'];
+
+        shouldReportNoDiagnostics(meta);
+
+        // JS exports
+        expect(exports.stVars.varA, `varA JS export`).to.eql(`a-val`);
+        expect(exports.stVars.varB, `varB JS export`).to.eql(`a-val`);
+    });
+    it(`should resolve cyclic vars`, () => {
+        const { sheets } = testStylableCore(`                
+            :vars {
+                /* @transform-warn(varA) ${STVar.diagnostics.CYCLIC_VALUE([
+                    `/entry.st.css: varB`,
+                    `/entry.st.css: varA`,
+                    `/entry.st.css: varB`,
+                ])} */
+                varA: a(value(varB));
+                
+                /* @transform-warn(varB) ${STVar.diagnostics.CYCLIC_VALUE([
+                    `/entry.st.css: varA`,
+                    `/entry.st.css: varB`,
+                    `/entry.st.css: varA`,
+                ])} */
+                varB: b(value(varA));
+            }
+            .root {
+                /* @decl(varA) prop: a(b(value(varA))) */
+                prop: value(varA);
+                
+                /* @decl(varB) prop: b(a(value(varB))) */
+                prop: value(varB);
+            }
+        `);
+
+        const { meta, exports } = sheets['/entry.st.css'];
+
+        // symbols
+        expect(STVar.get(meta, `varA`), `varA symbol`).to.contain({
+            text: `a(value(varB))`,
+        });
+        expect(STVar.get(meta, `varB`), `varB symbol`).to.contain({
+            text: `b(value(varA))`,
+        });
+
+        // exports
+        expect(exports.stVars, `JS exports`).to.eql({
+            varA: `a(b(a(value(varB))))`,
+            varB: `b(a(b(value(varA))))`,
+        });
+    });
+    it.skip(`should provide valid var path introspection - value(var, ...path)`);
```

Comment:
move to top

Reply:
moved to bottom until it will be implemented